base/REPLCompletions.jl

```diff
@@ -134,16 +134,68 @@ function complete_path(path::AbstractString, pos)
     return matches, nextind(path, pos - sizeof(prefix) - length(matchall(r" ", prefix))):pos, length(matches) > 0
 end
 
-function complete_methods(input::AbstractString)
-    tokens = split(input, '.')
-    fn = Main
-    for token in tokens
-        sym = symbol(token)
-        isdefined(fn, sym) || return UTF8String[]
-        fn = fn.(sym)
+# Determines whether a string endswiths ',' or '(' when disregarding whitespace_chars
+function ends_c(s::AbstractString)
+    for c in reverse(s)
+        if c in [',', '(']
+            return true
+        elseif !(c in whitespace_chars)
+            return false
+        end
     end
-    isgeneric(fn) || return UTF8String[]
-    UTF8String[string(m) for m in methods(fn)]
+    false
+end
+
+# Returns a range that includes the method name in front of the first non
+# closed start brace from the end of the string.
+function find_start_brace(s::AbstractString)
+    braces = 0
+    r = RevString(s)
+    i = start(r)
+    while !done(r, i)
+        c, i = next(r, i)
+        if c == '('
+            braces += 1
+        elseif c == ')'
+            braces -= 1
+        end
+        braces == 1 && break
+    end
+    braces != 1 && return 0:-1
+    startind = nextind(s, rsearch(s, non_identifier_chars, reverseind(r, i)))
+    return startind:endof(s)
+end
+
+# Return the next value when the sym is defined in current namespace fn
+function return_val(sym::Expr, fn)
```

Comment:
Both the comment and the function name are again unhelpful in describing what this does.